src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -114,11 +117,13 @@ void sl_ble_event_handling_task(void)
     // Application event map
     while (1)
     {
-        //! This semaphore is waiting for next ble event task
-        rsi_semaphore_wait(&sl_ble_event_sem, 0);
-
         // checking for events list
         event_id = rsi_ble_app_get_event();
+        if (event_id == -1)
+        {
+            rsi_semaphore_wait(&sl_ble_event_sem, 0);
+            continue;
+        }
```

Comment:
I think this change makes the code less clear. I would use the initial implementation. if the event id is `-1` or `unknown`, it is managed by the switch-case default.

Reply:
Why would we wait only after an event_id -1? Does `rsi_ble_app_get_event` block the task on its own? If it doesn't this task will continuously be active until a event_id -1.
 If `rsi_ble_app_get_event` does block. Why do we need a semaphore wait after a -1 event

Also what is a event_id -1, it's an error? an unhandled event? It is missing either a relevant name or a clear comment describing what you are doing here

Reply:
when we did not receive any event then this AP (rsi_ble_app_get_event ) will return -1 then we will wait till we get the event after getting the event we will continue. as @mkardous-silabs said and I'm also thinking the same initial implementation is better than this so updated the file